test/CMakeLists.txt

```diff
@@ -16,7 +16,15 @@ set (ranlux_24_48_test.pass_timeout_debug "900") # 15min
 set (ranlux_24_48_test.pass_timeout_release "720") # 12min
 
 # disable fast math for ALL TESTS
-check_cxx_compiler_flag("-fno-fast-math" _fno_fast_math_option)
+foreach(_fno_fast_math_flag -fno-fast-math /clang:-fno-fast-math)
+    string(MAKE_C_IDENTIFIER ${_fno_fast_math_flag} FLAG_DISPLAY_NAME)
+    check_cxx_compiler_flag(${_fno_fast_math_flag} ${FLAG_DISPLAY_NAME}_option)
+    if (${FLAG_DISPLAY_NAME}_option)
+        target_compile_options(oneDPL INTERFACE ${_fno_fast_math_flag})
```

Comment:
I'm a little confused about the logic here. From what I can tell, this is the effect based on which flag is present or not present in the compiler:

|  `-fno-fast-math` | `/clang:-fno-fast-math` | Behavior |
|---|---|---|
| Present | Not Present | `target_compile_options(oneDPL INTERFACE -fno-fast-math)` and `add_compile_options("-fno-fast-math")`  |
|  Not Present | Present | `target_compile_options(oneDPL INTERFACE /clang:-fno-fast-math)` and `add_compile_options("-fno-fast-math")`  |
| Present | Present | `target_compile_options(oneDPL INTERFACE -fno-fast-math)` and `add_compile_options("-fno-fast-math")`  |
| Not Present | Not Present | -  |

For the case where `/clang:-fno-fast-math` is available but `-fno-fast-math` is not, it seems like it will still try to set `-fno-fast-math` in line 21. Is that correct?

Reply:
Sorry this is a bug.  I'll update the code to properly use the caught option.